packages/winpanda/extra/src/winpanda/tests/unit/test_common_utils.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+import mock
+
+from common import utils
+from common.exceptions import ExternalCommandError
+from subprocess import SubprocessError, CalledProcessError
+
+
+@mock.patch('common.utils.SmartDL')
+def test_download(*args):
+    """
+    Download fake content and save it into file.
+    """
+    path = utils.download('', 'path')
+    assert isinstance(path, str)
+
+
+@mock.patch('os.path.exists', return_value=False)
+@mock.patch('os.mkdir')
+def test_unpack_unavailable_tar_path(*args):
+    """
+    Unpack wrong tarfile path.
+    """
+    with pytest.raises(ValueError):
+        utils.unpack('', '')
+
+
+@mock.patch('tarfile.open')
+@mock.patch('os.path.exists', return_value=True)
+def test_unpack(mock_tar, *args):
+    """
+    Check is unpack file result provide valid location.
+    """
+    mock_tar.__enter__.return_value = mock.Mock()
+    location = utils.unpack('', '')
+    assert isinstance(location, str)
```

Comment:
The "location" should represent an absolute path.

Reply:
replaced by 'endswith' because of absolute path.

Reply:
redesigned